Update components documentation for best practices and consistency

yadavmangesh07 opened the pull request:
## Pull Request Overview

This pull request updates the documentation for React components to reflect the latest best practices and features introduced in recent React versions. The updates include:

- Adding examples to demonstrate component usage.
- Clarifying prop usage for better understanding.
- Ensuring consistency across all components documentation.

## Related Issue

This addresses [Issue #24](https://github.com/AdityaSrivastava185/Reactjs-For-Beginners/issues/24).

## Checklist

- [x] Documentation updated
- [x] Examples added
- [x] Prop usage clarified
- [x] Consistency ensured across documentation

AdityaSrivastava185 commented on Components/Components.md:
```diff
@@ -1,141 +1,313 @@
 - [Component Lifecycle](#component-lifecycle)
 - [Render Props and Higher-Order Components (HOCs)](#render-props-and-higher-order-components-hocs)
 - [Context API for Prop Drilling](#context-api-for-prop-drilling)
-- [Concurrent Mode and Suspense](#concurrent-mode-and-suspense)
```
Remove this content because it is the higher hooks covered in the hooks topics seperately

AdityaSrivastava185 commented on Components/Components.md:
```diff
@@ -1,141 +1,313 @@
 - [Render Props and Higher-Order Components (HOCs)](#render-props-and-higher-order-components-hocs)
 - [Context API for Prop Drilling](#context-api-for-prop-drilling)
-- [Concurrent Mode and Suspense](#concurrent-mode-and-suspense)
+- [Concurrent Rendering and Suspense](#concurrent-rendering-and-suspense)
```
Please also remove this section from the docs

AdityaSrivastava185 commented on Components/Components.md:
```diff
@@ -1,141 +1,313 @@
 - [Context API for Prop Drilling](#context-api-for-prop-drilling)
-- [Concurrent Mode and Suspense](#concurrent-mode-and-suspense)
+- [Concurrent Rendering and Suspense](#concurrent-rendering-and-suspense)
 - [Memoization with `React.memo` and `useMemo`](#memoization-with-reactmemo-and-usememo)
```
Memoization is the react memo which is considered as a hook , the guide should be not confusing between the hooks and the components

AdityaSrivastava185 commented on Components/Components.md:
```diff
@@ -1,141 +1,313 @@
+
+function Welcome({ name }) {
+  return <h1>Hello, {name}!</h1>;
 }
```
Please explain the props used inside the component for clear and better understanding

AdityaSrivastava185 commented on Components/Components.md:
````diff
@@ -1,141 +1,313 @@
+useDebugValue(value);
+```
+## New Hooks in React 18
+React 18 introduces new hooks like 
````
Can you please remove from here

AdityaSrivastava185 commented on Components/Components.md:
````diff
@@ -1,141 +1,313 @@
+startTransition(() => {
+  setState(newState);
+});
+```
````
Till here

AdityaSrivastava185 reviewed (changes requested):
Thank you for the contribution but there is still some changes we would like to convey ,Please check the changes requests ✅

AdityaSrivastava185 commented:
I have assigned you the issue

yadavmangesh07 commented:
This pull request includes significant updates to the `Components/Components.md` file, focusing on improving the structure and content related to React components. The most important changes include restructuring the table of contents, enhancing descriptions of functional and class components, and adding detailed explanations of component lifecycle and nested components.

### Documentation Improvements:

* **Table of Contents Update:**
  - Reorganized the table of contents to better reflect the updated sections and topics.

* **Enhanced Component Descriptions:**
  - Improved descriptions and examples for functional and class components, emphasizing the benefits of functional components with hooks.
  - Added explanations on the advantages of functional components over class components.

* **Component Lifecycle Explanation:**
  - Added a detailed section on the component lifecycle, including mounting, updating, and unmounting phases with examples.

* **Nested Components and Component Tree:**
  - Introduced a new section on component tree and nested components, explaining how to create and manage nested components with examples.

* **Export and Import Components:**
  - Added a section on exporting and importing components, providing examples to demonstrate modular code organization.

yadavmangesh07 commented:
Hi @AdityaSrivastava185,

I hope you're doing well. I am writing to kindly request that you add the "hacktoberfest-accepted" label to my pull request [#31](https://github.com/AdityaSrivastava185/Reactjs-For-Beginners/pull/31) that was recently merged.

Thank you for your help!

Best regards,
@yadavmangesh07
